### PR TITLE
Update BBE descriptions and comments

### PR DESCRIPTION
**Pull request by suleka96**
## Purpose
$title

### CODE REVIEW

File: examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal [hunk 1]
```diff
@@ -8,10 +8,8 @@ type Coord record {
 public function main() {
     json j = {x: 1.0, y: 2.0};
 
-    // With mutable values, it would not be type-safe to allow a cast.
-    // Cast to `T` will work on the mutable structure `s` only if the inherent type
-    // of `s` is a subtype of `T`.
-    // Casting of immutable value will work.
+    // Use `cloneReadOnly()` to create a read-only copy of the mutable value.
+    // Then you can successfully cast the resulting immutable value.
```
Comment (praneesha) [inline examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal:12]:
```suggestion
    // Then, you can cast the resulting immutable value successfully.
```

File: examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.description [hunk 2]
```diff
@@ -1,5 +1,5 @@
-// With mutable values, would not be type-safe to allow a cast. 
+// With mutable values, it would not be type-safe to allow a cast.
 // Mutable structures have the `inherent` type that limits mutation.
 // Cast to `T` will work on mutable structure `s` only if the `inherent` type
 // of `s` is a subtype of `T`.
-// Casting of immutable value will work but it does not do numeric conversions.
+// Casting immutable values will work, but it does not do numeric conversions.
```
Comment (praneesha) [inline examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.description:5]:
```suggestion
// Casting immutable values will work but it does not do numeric conversions.
```

File: examples/converting-to-user-defined-type/converting_to_user_defined_type.description [hunk 3]
```diff
@@ -2,7 +2,7 @@
 // module can be used to convert a value to a user-defined type.
 // Result recursively uses specified type as inherent type of new value.
 // Automatically performs numeric conversions as necessary.
-// Every part of value is cloned, including immutable structural values.
+// Every part of the value is cloned, including immutable structural values.
```
Comment (praneesha) [inline examples/converting-to-user-defined-type/converting_to_user_defined_type.description:5]:
```suggestion
// Every part of the value is cloned including immutable structural values.
```

**Review by praneesha — APPROVED** [review]

File: examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal [hunk 4]
```diff
@@ -8,10 +8,8 @@ type Coord record {
 public function main() {
     json j = {x: 1.0, y: 2.0};
 
-    // With mutable values, it would not be type-safe to allow a cast.
```
Comment (MaryamZi) [inline examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal:11]: Why was this removed?

File: examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal [hunk 5]
```diff
@@ -8,10 +8,8 @@ type Coord record {
...
-    // Cast to `T` will work on the mutable structure `s` only if the inherent type
-    // of `s` is a subtype of `T`.
-    // Casting of immutable value will work.
+    // Use `cloneReadOnly` to create a read-only copy of the mutable value.
```
Comment (MaryamZi) [inline examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal:11]:
```suggestion
    // Use `cloneReadOnly()` to create a read-only copy of the mutable value.
```

File: examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.description [hunk 6]
```diff
@@ -1,5 +1,5 @@
-// With mutable values, would not be type-safe to allow a cast. 
+// With mutable values, it would not be type-safe to allow a cast.
 // Mutable structures have the `inherent` type that limits mutation.
 // Cast to `T` will work on mutable structure `s` only if the `inherent` type
 // of `s` is a subtype of `T`.
-// Casting of immutable value will work but it does not do numeric conversions.
+// Casting immutable values will work but it does not do numeric conversions.
```
Comment (MaryamZi) [inline examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.description:5]:
```suggestion
// Casting immutable values will work but it does not do numeric conversions.

```

File: examples/query-expressions/query_expressions.description [hunk 7]
```diff
@@ -1,2 +1,2 @@
-// Query-like expressions start with `from` clause and end with `select` clause. It is a
+// Query-like expressions start with the `from` clause and end with the `select` clause. It is a
```
Comment (MaryamZi) [inline examples/query-expressions/query_expressions.description:1]:
```suggestion
// Query-like expressions start with a `from` clause and end with a `select` clause. It is a
```

File: examples/table-syntax/table_syntax.bal [hunk 8]
```diff
@@ -5,7 +5,7 @@ type Employee record {
     int salary;
 };
 
-// Creates a `table` with `Employee` type members, where each
+// Creates a `table` with members of `Employee` type, where each
```
Comment (MaryamZi) [inline examples/table-syntax/table_syntax.bal:8]:
```suggestion
// Creates a `table` with members of the `Employee` type, where each
```

File: examples/table-syntax/table_syntax.bal [hunk 9]
```diff
@@ -17,11 +17,10 @@ function increaseSalary(int n) {
     foreach Employee e in t {
         e.salary += n;
     }
-
 }
 
 public function main() {
-    // Retrieves `Employee` with key value `Fred`.
+    // Retrieves `Employee` with `Fred`, which is the value of the key.
```
Comment (MaryamZi) [inline examples/table-syntax/table_syntax.bal:23]:
```suggestion
    // Retrieves the `Employee` member with `"Fred"` as the value of the key.
```

File: examples/table-syntax/table_syntax.bal [hunk 10]
```diff
@@ -17,11 +17,10 @@ function increaseSalary(int n) {
     foreach Employee e in t {
         e.salary += n;
     }
-
```
Comment (lochana-chathura) [inline examples/table-syntax/table_syntax.bal:20]: I think this newline was intentional. I guess we mark the code segment related to the comment above, using this newline.  Without it both `}`  get captured under the comment.

File: examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal (hunk 4 above)
Comment (suleka96) [reply examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal:11]: The same piece of text was repeated in both the description and comment sections. So I kept the description and changed the comment according to the slides

File: examples/destructuring-records/destructuring_records.description [hunk 11]
```diff
@@ -1,3 +1,3 @@
-// Destructuring records is particularly useful with `query expressions`, but works anywhere you can have `var`.
+// Destructuring records is particularly useful with `query expressions` but works anywhere you can have `var`.
```
Comment (pubudu91) [inline examples/destructuring-records/destructuring_records.description:1]:
```suggestion
// Destructuring records is particularly useful with query expressions but works anywhere you can have `var`.
```
Since "query expressions" is not code

File: examples/structured-keys/structured_keys.description [hunk 12]
```diff
@@ -1,3 +1,3 @@
-// Key fields can be structured where they belong to any subtype of plain data. The Value of key field
-// must be immutable. Initializer of `readonly` field will be constructed as immutable. In other cases,
+// Key fields can be structured in places where they belong to any subtype of plain data. The value of the key field
```
Comment (pubudu91) [inline examples/structured-keys/structured_keys.description:1]: What does "Key fields can be structured in places" mean? 🤔  From how it was originally phrased, I understood it to mean that key fields can be of structured types which are subtypes of `anydata`. I'm didn't quite get the new phrasing 🤔

Comment (suleka96) [reply examples/structured-keys/structured_keys.description:1]: Oh the original phrase `Key fields can be structured where they belong to any subtype of plain data.` in a literal sense read to me somewhat like:

> key fields can be structured types wherever, as long they belong to any subtype of plain data.

 That's why I added the `in places ` part for it to read better.

Comment (pubudu91) [reply examples/structured-keys/structured_keys.description:1]: I see. @MaryamZi thoughts? 🤔

File: examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal (hunk 4 above)
Comment (MaryamZi) [reply examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal:11]: When someone copies the example and runs it separately they won't see the description? Better to continue to have it IMO.

File: examples/structured-keys/structured_keys.description (hunk 12 above)
Comment (MaryamZi) [reply examples/structured-keys/structured_keys.description:1]: How about

```suggestion
// Key fields can be structured if they belong to any subtype of plain data. The value of the key field
```

**Review by pubudu91 — APPROVED** [review]

File: examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal [hunk 13]
```diff
@@ -7,11 +7,11 @@ type Coord record {
 
 public function main() {
     json j = {x: 1.0, y: 2.0};
-
+    
+
```
Comment (MaryamZi) [inline examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal:11]:
```suggestion

```

Comment (MaryamZi) [inline examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal:11]:
```suggestion
    
```

File: examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal [hunk 14]
```diff
@@ -7,11 +7,11 @@ type Coord record {
 
 public function main() {
     json j = {x: 1.0, y: 2.0};
-
+    
+    
     // With mutable values, it would not be type-safe to allow a cast.
-    // Cast to `T` will work on the mutable structure `s` only if the inherent type
-    // of `s` is a subtype of `T`.
```
Comment (MaryamZi) [inline examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal:13]: Why is this removed?

File: examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal [hunk 15]
```diff
@@ -7,11 +7,11 @@ type Coord record {
...
-    // Casting of immutable value will work.
+    // Use `cloneReadOnly()` to create a read-only copy of the mutable value.
+    // Then, you can cast the resulting immutable value successfully.
```
Comment (MaryamZi) [inline examples/converting-from-json-to-user-defined-type/converting_from_json_to_user_defined_type.bal:14]:
```suggestion
    // Cast to `T` will work on the mutable structure `s` only if the inherent type
    // of `s` is a subtype of `T`.
    // Use `cloneReadOnly()` to create a read-only copy of the mutable value.
    // Then, you can cast the resulting immutable value successfully.
```

File: examples/structured-keys/structured_keys.description [hunk 16]
```diff
@@ -1,3 +1,3 @@
-// Key fields can be structured where they belong to any subtype of plain data. The Value of key field
-// must be immutable. Initializer of `readonly` field will be constructed as immutable. In other cases,
-// can use `cloneReadOnly` to create an immutable value.
+// Key fields can be structured if they belong to any subtype of plain data. The value of the key field
```
Comment (MaryamZi) [inline examples/structured-keys/structured_keys.description:1]: On second thought should this be

```suggestion
// Key fields can be structured as long as they belong to any subtype of plain data. The value of the key field
```

**Review by MaryamZi — APPROVED** [review]
